### DISCUSSION TITLE
Does Shadow support Gossipsub?

### DISCUSSION THREAD

**Discussion by srw** [shadow/shadow#3303] [Q&A]
Hi,

I was following [this](https://discuss.libp2p.io/t/rough-stress-metrics-for-gossipsub/2223/3) thread and I wonder if Shadow supports Gossipsub for simulations?

Thanks!

**Comment by sporksmith** [comment]
It looks like there's an MR to add tests for it, but it hasn't been merged yet: https://github.com/shadow/shadow/pull/3242

It also looks like someone has successfully run simulations with it in the past: https://github.com/shadow/shadow/discussions/2622

**Reply by srw** [reply]
Very interesting, could you clarify why they are using a simulation approach consuming real computer resources instead of just using graph theory and adding bandwidth and delay data for the edges? Am I missing something or this could also be simulated using basic graph theory but without consuming a lot of real resources?

**Reply by sporksmith** [reply]
I'm not very familiar with gossipsub or what they were doing with it. In general there are many potential reasons though; some touched on by the thread you originally linked. e.g. it can take substantial human work to model a complex piece of software, and can be difficult to be certain that the model actually sufficiently reflects the implementation of the real application. See also https://github.com/shadow/shadow?tab=readme-ov-file#why-is-shadow-needed. (Or for much longer answers, the papers and talks linked at the bottom of https://shadow.github.io/)
